### PR TITLE
Cleanup rafthttp

### PR DESCRIPTION
**Pull request by xiang90**
/cc @yichengq @kelseyhightower @barakmich

### CODE REVIEW

File: rafthttp/transport.go [hunk 1]
```diff
@@ -35,12 +35,25 @@ type Raft interface {
 }
 
 type Transporter interface {
+	// Handler returns the HTTP handler of the transporter.
+	// A transporter HTTP handler handles the HTTP requests
+	// from remote peers.
```
Comment (yichengq) [inline rafthttp/transport.go]: and you should mount it on '/raft' endpoint

Comment (xiang90) [inline rafthttp/transport.go]: do we have a const for this?

Comment (yichengq) [inline rafthttp/transport.go]: yes. RaftPrefix

File: rafthttp/transport.go [hunk 2]
```diff
@@ -35,12 +35,32 @@ type Raft interface {
...
+	// The handler MUST be used to handle RaftPrefix(/raft)
+	// endpoint.
 	Handler() http.Handler
+	// Send sends out the given messages to the remove peers.
```
Comment (yichengq) [inline rafthttp/transport.go]: s/remove/remote

**Comment by yichengq** [comment]
lgtm after addressing comment

**Pull request merged**